sound/soc/sof/pcm.c

```diff
@@ -486,10 +486,17 @@ static int sof_pcm_open(struct snd_pcm_substream *substream)
 	spcm->prepared[substream->stream] = false;
 
 	ret = snd_sof_pcm_platform_open(sdev, substream);
-	if (ret < 0)
+	if (ret < 0) {
 		dev_err(sdev->dev, "error: pcm open failed %d\n", ret);
+		return ret;
+	}
 
-	return ret;
+	/* update active pcm counts */
+	sdev->pcm_active++;
+	if (spcm->stream[substream->stream].avail_at_lp)
+		sdev->lp_pcm_active++;
```

Comment:
don't we need a mutex here?

Reply:
We don't need mutex or lock as it is used in open/close which is already managed by ALSA framework.

Reply:
OK, will add it.